tensorflow_probability/python/positive_semidefinite_kernels/dot_product_test.py

```diff
@@ -0,0 +1,51 @@
+# Copyright 2018 The TensorFlow Probability Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ============================================================================
+"""Tests for Linear."""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+from absl.testing import parameterized
+
+import numpy as np
+import tensorflow as tf
+
+from tensorflow_probability import positive_semidefinite_kernels as psd_kernels
+
+
+class LinearTest(tf.test.TestCase, parameterized.TestCase):
+
+  def testMismatchedFloatTypesAreBad(self):
+    psd_kernels.Linear(1, 1)  # Should be OK (float32 fallback).
+    psd_kernels.Linear(np.float32(1.), 1.)  # Should be OK.
+    with self.assertRaises(TypeError):
+      psd_kernels.Linear(np.float32(1.), np.float64(1.))
+
+  def testValidateArgs(self):
+    k = psd_kernels.Linear(-1., -1., validate_args=True)
+    with self.assertRaises(tf.errors.InvalidArgumentError):
+      self.evaluate(k.amplitude)
+
+    with self.assertRaises(tf.errors.InvalidArgumentError):
+      self.evaluate(k.length_scale)
+
+    # But `None`'s are ok
+    k = psd_kernels.Linear(None, None, validate_args=True)
+    self.evaluate(k.apply([1.], [1.]))
+
```

Comment:
I would add a few more tests here. Following https://github.com/tensorflow/probability/blob/master/UNITTEST.md, it would be good to make these tests dtype dependent (one issue we have encountered in the past is float32 working but not float64 working for methods).

I think the two other main types of tests that should be here are correctness, and testing shapes of the form [B.... F...]. https://github.com/tensorflow/probability/blob/master/tensorflow_probability/python/positive_semidefinite_kernels/exp_sin_squared_test.py#L50 should be a good guide on what type of test can be added.

Reply:
Added tests for float64 as well as float32.
Added tests for shapes.